core/parser/src/main/java/com/blazebit/persistence/impl/SimpleQueryGenerator.java

```diff
@@ -535,7 +535,7 @@ public void visit(FunctionExpression expression) {
         BooleanLiteralRenderingContext oldBooleanLiteralRenderingContext = setBooleanLiteralRenderingContext(BooleanLiteralRenderingContext.PLAIN);
         ParameterRenderingMode oldParameterRenderingMode = setParameterRenderingMode(ParameterRenderingMode.PLACEHOLDER);
         boolean hasExpressions = expression.getExpressions().size() > 0;
-        String functionName = expression.getFunctionName();
+        String functionName = expression.getFunctionName().toUpperCase();
```

Comment:
The `ResolvingQueryGenerator` is the _last bastion_ after which a query gets passed to a JPA provider so it is sufficient to handle it only there. Everything else is just intermediate processing.